### PR TITLE
Refactor promote command in favor of publish

### PR DESCRIPTION
**Pull request by mtojek**
This PR refactors "promote" command in favor of future "publish" command, which will reuse the storage part.

Extracted from: https://github.com/elastic/elastic-package/pull/292 to make the PR reviewable.

### CODE REVIEW

File: internal/cobraext/const.go [hunk 1]
```diff
@@ -36,8 +36,8 @@ const (
 	ReportOutputFlagName        = "report-output"
 	ReportOutputFlagDescription = "output location for test report"
 
-	VerboseFlagName        = "verbose"
-	VerboseFlagDescription = "verbose mode"
+	SkipPullRequestFlagName        = "skip-pull-request"
+	SkipPullRequestFlagDescription = "skip opening a new pull request"
```
Comment (ycombinator) [inline internal/cobraext/const.go:40]: Where are these two used? I think maybe these are not supposed to be in this PR?

Comment (mtojek) [reply internal/cobraext/const.go]: You're right, didn't clean it properly.

File: internal/storage/storage.go [hunk 2]
```diff
@@ -39,16 +42,44 @@ type PackageVersion struct {
 	Name    string
 	Version string
 
+	root   string
```
Comment (ycombinator) [inline internal/storage/storage.go:45]: I didn't understand why this is needed for promotion.

Comment (mtojek) [reply internal/storage/storage.go]: Same reason, missed during extraction. Fixed.

**Review by ycombinator — APPROVED** [review]
Refactoring LGTM. Thanks for extracting it out into its own PR — made it a lot easier to review.

We should get these changes into the `integrations` repo and let a few promotions happen with them before merging the second PR. WDYT?

**Comment by mtojek** [comment]
> We should get these changes into the integrations repo and let a few promotions happen with them before merging the second PR. WDYT?

Actually this part is not actively used by elastic/integrations. It's more about running `elastic-package promote`, but it means that users have to download and use the latest version. I think in this situration we can take this risk and push the `publish` command. I don't think we'll get an immediate feedback.

**Comment by mtojek** [comment]
The CI job go stuck for some reason. I will merge this one.

**Comment by ycombinator** [comment]
> Actually this part is not actively used by elastic/integrations. It's more about running `elastic-package promote`, but it means that users have to download and use the latest version. I think in this situration we can take this risk and push the `publish` command. I don't think we'll get an immediate feedback.

Ah yes, I forgot, `elastic-package promote` is a manual step. I agree, we can take the risk.

**Pull request merged**